src/handlers/askingArticleSubmissionReason.js

```diff
@@ -1,46 +1,113 @@
-        altText:
-          '若要確認請輸入「y」、若要放棄請輸入「n」、若要重新填寫理由請輸入「r」',
-        template: {
-          type: 'buttons',
-          text: '請確認',
-          actions: [
-            createPostbackAction('明白我要送出', 'y', issuedAt),
-            createPostbackAction('重寫送出的理由', 'r', issuedAt),
-            createPostbackAction('放棄送出', 'n', issuedAt),
-          ],
+        type: 'flex',
+        altText: text,
+        contents: {
+          type: 'bubble',
+          styles: {
+            footer: {
+              separator: true,
+            },
+          },
+          body: {
+            type: 'box',
+            layout: 'vertical',
+            contents: [
+              {
+                type: 'text',
+                text: '把訊息送進資料庫',
+                weight: 'bold',
+                color: '#1DB446',
+                size: 'sm',
+              },
+              {
+                type: 'separator',
+                margin: 'xxl',
+              },
+              {
+                type: 'text',
+                text: '訊息',
+                weight: 'bold',
+                size: 'xl',
+                margin: 'xl',
+              },
+              {
+                type: 'text',
+                text: data.searchedText,
+                size: 'xs',
+              },
+              {
+                type: 'separator',
+                margin: 'xxl',
+              },
+              {
+                type: 'text',
+                text: '覺得是謠言的理由',
+                weight: 'bold',
+                size: 'xl',
+                margin: 'xl',
+              },
+              {
+                type: 'text',
+                text: reason,
+                size: 'xxs',
+              },
+              {
+                type: 'separator',
+                margin: 'xxl',
+              },
+              {
+                type: 'text',
+                text:
+                  '若您送出的訊息或理由意味不明、造成闢謠編輯的困擾，未來您將無法送出文章。',
+                color: '#ff0000',
+                wrap: true,
+                margin: 'xxl',
+              },
+              {
+                type: 'button',
+                margin: 'xl',
+                style: 'primary',
+                action: createPostbackAction('放棄送出', 'n', issuedAt),
+              },
+              {
+                type: 'button',
+                action: createPostbackAction('重寫送出的理由', 'r', issuedAt),
+              },
+              {
+                type: 'button',
+                action: createPostbackAction('明白我要送出', 'y', issuedAt),
+              },
+            ],
+          },
         },
       },
     ];
     data.reasonText = reason;
     state = 'ASKING_ARTICLE_SUBMISSION';
-  } else {
+  } else if (event.input !== REASON_PLACEHOLDER) {
     // Track whether user create Article or not if the Article is not found in DB.
     ga(userId, { ec: 'Article', ea: 'Create', el: 'No' });
 
     replies = [{ type: 'text', text: '訊息沒有送出，謝謝您的使用。' }];
     state = '__INIT__';
+  } else {
```

Comment:
1. If we use `if (event.input === REASON_PLACEHOLDER)` at the beginning of this `if-else if-else` block, the the `else-if` and `else` won't need to test for `REASON_PLACEHOLDER`, which is more readable.
2. This logic should be implemented in `askingReplyRequestReason()` as well

Reply:
In `askingReplyRequestReason()`, I don't think we need to ask users to revise their reason if `event.input === REASON_PLACEHOLDER` because I think this means the user doesn't want to tell us about his reason if any. Since it's not compulsory, I'll just take `REASON_PLACEHOLDER` as `n`.

Reply:
Thanks for the explanation! It makes sense if we take `REASON_PLACEHOLDER` as `n` in `askingReplyRequestReason()`.